### ISSUE TITLE
Outdated information on ServiceAccount tokens in "Managing Service Accounts" page

### ISSUE THREAD

**Issue by nnlkcncff** [kubernetes/website#43111]
> https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#create-token
> https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#delete-token

According to [Concepts/Configuration/Secrets](https://kubernetes.io/docs/concepts/configuration/secret/#service-account-token-secrets) since version 1.22 using the manifest from the section does not automatically generate anything:
```yaml
apiVersion: v1
kind: Secret
type: kubernetes.io/service-account-token
metadata:
  name: mysecretname
  annotations:
    kubernetes.io/service-account.name: myserviceaccount
```

The secret will not be created and will not appear in the service account configuration.
Additionally, the `LegacyServiceAccountTokenNoAutoGeneration` feature gate is [GA](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#urgent-upgrade-notes), so the described behaviour cannot be reproduced at this time.

Thus sections "Create additional API tokens" and "Delete/invalidate a ServiceAccount token" should be updated.

**Comment by dipesh-rawat** [comment]
/retitle Outdated information on ServiceAccount tokens in "Managing Service Accounts" page

**Comment by nnlkcncff** [comment]
Hi @Shubham82

Sorry, it's my fault — my secret wasn't created because I misspelled the service account name, but got the message "secret/mysecretname created" (I've experimented with the default service account), so [Create additional API tokens](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#create-token) section is correct.

But there is still a discrepancy with [Delete/invalidate a ServiceAccount token](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#delete-token) section; because of `LegacyServiceAccountTokenNoAutoGeneration` future gate is GA there is no way to turn it off, therefore, the new secret name (and selfLink) will not be written to the service account configuration, nor will it be generated and written to the configuration automatically after the secret is deleted.

```bash
kubectl apply -f - <<EOF
apiVersion: v1
kind: ServiceAccount
metadata:
  name: build-robot
EOF

kubectl apply -f - <<EOF
apiVersion: v1
kind: Secret
metadata:
  name: build-robot-secret
  annotations:
    kubernetes.io/service-account.name: build-robot
type: kubernetes.io/service-account-token
EOF

kubectl get serviceaccount build-robot -o yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","kind":"ServiceAccount","metadata":{"annotations":{},"name":"build-robot","namespace":"default"}}
  creationTimestamp: "2023-09-20T08:08:27Z"
  name: build-robot
  namespace: default
  resourceVersion: "1436161"
  uid: 14340263-39c2-404d-8cd5-5ca8aaa0cba6
```

There are no similar lines in the service account configuration, as in [the example](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#delete-token):
```yaml
metadata:
  selfLink: /api/v1/namespaces/examplens/serviceaccounts/example-automated-thing
secrets:
  - name: example-automated-thing-token-zyxwv
```

You can not reproduce exactly what was described in this section, because of that it should be updated.

**Comment by sftim** [comment]
/remove-kind support

This is not really a support request; it's a bug report.

**Comment by sftim** [comment]
/priority backlog

The remaining differences are very minor. I guess if we want to get those updated, we can do that.

**Comment by Shubham82** [comment]
@nnlkcncff I also tried it, for me it is also not showing the `secret` and `selfLink` fields here is outcome of `kubectl get as build-robot -o yaml command`

```
$ kubectl get sa build-robot -o yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","kind":"ServiceAccount","metadata":{"annotations":{},"name":"build-robot","namespace":"default"}}
  creationTimestamp: "2023-09-20T10:29:43Z"
  name: build-robot
  namespace: default
  resourceVersion: "2565243"
  uid: 6d3c80fc-b909-4caa-89d1-d0aa860fb97a
```

I also tried to deploy SA  with the annotation `kubernetes.io/enforce-mountable-secrets: "true"` as mentioned  [here](https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/service-account-v1/) but still it showing the same output.

```
$ kubectl get sa build-robot -n test2 -o yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","kind":"ServiceAccount","metadata":{"annotations":{"kubernetes.io/enforce-mountable-secrets":"true"},"name":"build-robot","namespace":"test2"}}
    kubernetes.io/enforce-mountable-secrets: "true"
  creationTimestamp: "2023-09-20T11:13:01Z"
  name: build-robot
  namespace: test2
  resourceVersion: "2567292"
  uid: f8627d83-5a3a-4efd-b083-53bc6c59b8d3
```

**Comment by Shubham82** [comment]
But the secret name is showing under the token field (as expected) when we describe the sa.

```
$ kubectl describe sa build-robot
Name:                build-robot
Namespace:           default
Labels:              <none>
Annotations:         <none>
Image pull secrets:  <none>
Mountable secrets:   <none>
Tokens:              build-robot-secret
Events:              <none>

```
```
$ kubectl describe sa build-robot -n test2
Name:                build-robot
Namespace:           test2
Labels:              <none>
Annotations:         kubernetes.io/enforce-mountable-secrets: true
Image pull secrets:  <none>
Mountable secrets:   <none>
Tokens:              mysecretname
Events:              <none>
```


Also, the secret field is always empty in both case:

```
kubectl get sa
NAME          SECRETS   AGE
build-robot   0         34m
```

```
$ kubectl get sa -n test2
NAME          SECRETS   AGE
build-robot   0         52m
```

**Comment by Shubham82** [comment]
> /remove-kind support
> 
> This is not really a support request; it's a bug report.

yeah, it is a bug, you're right.

I think, there is a need for improvement here, we can remove the `secret` and `selfLink` fields. but what about these empty Secrets fields while running `kubectl get sa` command?

**Comment by liggitt** [comment]
Manually created secrets do not appear in the service account `.secrets` list. That list was *only* for auto-generated secrets, so it is expected a manually created Secret would not appear in that list.

**Comment by liggitt** [comment]
/kind bug

readding the bug label since the delete content is not accurate, but the create content is correct as-is

**Comment by Shubham82** [comment]
> Manually created Secret API objects do not appear in the service account API object's `.secrets` field. That field is _only_ populated with auto-generated secrets, so it is expected a manually created Secret would not appear in that list.

Thanks @liggitt for clarifying things,  I'm also thinking the same for autogenerated secrets it worked but not with manually created Secret API.
can we mention this thing under [Manually create a long-lived API token for a ServiceAccount](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-a-long-lived-api-token-for-a-serviceaccount), so that it will be more understandable to the readers/users?

**Comment by Shubham82** [comment]
> > Manually created Secret API objects do not appear in the service account API object's `.secrets` field. That field is _only_ populated with auto-generated secrets, so it is expected a manually created Secret would not appear in that list.
> 
> Thanks @liggitt for clarifying things, I'm also thinking the same for autogenerated secrets it worked but not with manually created Secret API. can we also mention this thing under [Manually create a long-lived API token for a ServiceAccount](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-a-long-lived-api-token-for-a-serviceaccount), so that it will be more understandable to the readers/users?

@liggitt @sftim your thoughts on this.

**Comment by Shubham82** [comment]
I'm up for it
/assign

**Comment by Shubham82** [comment]
Fixed it here:  #43451
